test/testinput/bufr_snow_adpsfc.yaml

```diff
@@ -7,30 +7,42 @@ observations:
   - obs space:
       name: bufr
 
-      obsdatain: "./testinput/gdas.t06z.adpsfc.tm00.bufr_d"
+      obsdatain: "./testinput/gdas.t06z.adpsfc.tm00.bufr_d" 
+
       exports:
         variables:
           timestamp:
             datetime:
-              year: "*/YEAR[1]"
-              month: "*/MNTH[1]"
-              day: "*/DAYS[1]"
-              hour: "*/HOUR[1]"
-              minute: "*/MINU[1]"
+              year: "[NC000007/YEAR[1],*/YEAR]"
```

Comment:
@YoulongXia-NOAA and @rmclaren ,
Are we going to keep using the "NC******" from the BUFR files?
I thought we are trying to step away from these. This is just a question, not an opinion.

Reply:
> @YoulongXia-NOAA and @rmclaren , Are we going to keep using the "NC******" from the BUFR files? I thought we are trying to step away from these. This is just a question, not an opinion.

This is following Ron's code and suggestion to work for this yaml file. It is bufr rather than prebufr format. As NC000007 has YEAR[1] and YEAR[2], if we used */YEAR, the yaml file will not work.

Reply:
Ilianna is right, the query "*/YEAR[1]" should work fine as you are only ever interested in the first YEAR. YEAR is actually grouped with HOUR, MINU ... so on, so they all follow the same pattern. Sorry I didn't catch this

Most of the time you will not need to mention a specific subset in your queries, so you should avoid doing so. Once in a while you may end up with a case were there isn't a choice but to query specific subsets, but this is not one of those times...

Reply:
@ilianagenkova @rmclaren, I used   
              year: "[*/YEAR[1],*/YEAR]"
              month: "[*/MNTH[1],*/MNTH]"
              day: "[*/DAYS[1], */DAYS]"
              hour: "[*/HOUR[1], */HOUR]"
              minute: "[*/MINU[1], */MINU]"
to test adpsfc snow yaml file, and compared the nc output with those when NC000007 is used, the results are the same. This means that it works when I removed NC000007 as you suggested. @rmclaren, could you please modify my yaml file to remove NC000007? Thank you.

Reply:
Note it should have * before / for all variables. As I directly copy from my screen and past them here, some are missing. See my new yaml file
bufr_snow_adpsfcV2_newTest.yaml in   /scratch1/NCEPDEV/da/Youlong.Xia/test/test_fromRon. Youlong